src/main/java/lesson10Task1/PhoneBook.java

```diff
@@ -0,0 +1,79 @@
+package lesson10Task1;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+public class PhoneBook {
+
+    private Map<String, PhoneBookEntry> entries;
+
+    public PhoneBook() {
+        entries = new HashMap<>();
+    }
+
+    public void addEntry(String firstName, String lastName, String phoneNumber, String address) {
+        if (firstName.isEmpty() || lastName.isEmpty() || address.isEmpty()) {
+            throw new IllegalArgumentException("the address, first name, and last name fields are mandatory");
+        }
+
+
+        PhoneBookEntry entry = new PhoneBookEntry(firstName, lastName, phoneNumber, address);
+        String key = firstName.toLowerCase() + " " + lastName.toLowerCase();
```

Comment:
What if there are two people with name John Smith ? one will override another.

Reply:
Consider to use list as a collection of phonebookentries, then filter by name and surname